### PR TITLE
refactor: remove dead code, improve test coverage and fix convoluted client code

### PR DESCRIPTION
**Pull request by Jackmin801**
**Goals:**

- [x] Take out groundtruth code in client
- [x] Improve client code coverage
- [x] Take out kwargs logic in data to doc parsing because it creates silent failures

### CODE REVIEW

File: jina/clients/request/helper.py [hunk 1]
```diff
@@ -59,13 +58,9 @@ def _build_doc_from_content():
...
+def _add_docs(req: DataRequest, batch, data_type: DataInputType, _kwargs):
     da = DocumentArray()
     for content in batch:
-        if isinstance(content, tuple) and len(content) == 2:
-            d, data_type = _new_doc_from_data(content[0], data_type, **_kwargs)
-            da.append(d)
-        else:
-            d, data_type = _new_doc_from_data(content, data_type, **_kwargs)
-            da.append(d)
+        d, data_type = _new_doc_from_data(content, data_type, **_kwargs)
```
Comment (JoanFM) [inline jina/clients/request/helper.py:64]: u are removing these `_kwargs` I think it is not needed to spend time on this

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/clients/request/helper.py (hunk 1 above)
Comment (Jackmin801) [reply jina/clients/request/helper.py:64]: I will try remove them and see if they still pass tests. But it is possible to access the _kwargs here.

The _kwargs I removed are in another function where it is created but never used

Comment (Jackmin801) [reply jina/clients/request/helper.py]: It technically might be used here. But it seems to never need it in the tests.
https://github.com/jina-ai/jina/blob/master/jina/clients/request/helper.py#L37-L38

**Comment by Jackmin801** [comment]
The groundtruth test somehow fails when kwargs are removed; not when the condition that handles kwargs was removed

**Comment by Jackmin801** [comment]
Mystery Solved
---
With kwargs on, when _new_doc_from_data is entered it has these parameters:
```python
data = (<Document ('id',) at...9d321ea9c>, <Document ('id',) at...c1263d4c1>)
data_type = <DataInputType.DOCUMENT: 1> | <DataInputType.AUTO: 0> # The first pair hits the auto the subsequent pairs have DOCUMENT because of the first pair setting the data_type
kwargs = {'proxy': False, 'host': '0.0.0.0', 'port': 12345, 'tls': False, 'asyncio': False, 'tracing': False, 'traces_exporter_host': None, 'traces_exporter_port': None, 'metrics': False, 'metrics_exporter_host': None, 'metrics_exporter_port': None, 'protocol': <GatewayProtocolType.GRPC: 0>, 'prefetch': 1000}
```

In this case, the kwargs are interpreted as tags and the Document constructor ignores that the data doesnt make sense
```python
Document(data, **kwargs) = <Document ('id', 'tags') at 10ab62829cef221b0712a30dfe174ad3>
```

Without the kwargs, it realizes the data makes no sense
```python
ValueError("Failed to initialize docarray.document.Document from obj=(<Document ('id',) at 0da4c26c7cc2909ee7f7c589d321ea9c>, <Document ('id',) at 61558bc5fe3d78093abf840c1263d4c1>), kwargs={}")
```

**Comment by JoanFM** [comment]
what is the test with groundtruth? this should be rmeoved

**Comment by Jackmin801** [comment]
https://github.com/jina-ai/jina/blob/master/tests/integration/v2_api/test_func_routing.py#L23

Yup. I've converted it to not use the groundtruth tuple.

File: jina/clients/request/helper.py (hunk 1 above)
Comment (Jackmin801) [reply jina/clients/request/helper.py]: I think it's a good idea to remove it. As described in https://github.com/jina-ai/jina/pull/5568#issuecomment-1369785922, it creates weird silent failures.

**Review by JoanFM — APPROVED** [review]

**Pull request merged**